src/main/java/org/zaproxy/zap/extension/jwt/attacks/SignatureAttack.java

```diff
@@ -143,6 +146,43 @@ private boolean executePubliclyWellKnownHMacSecretAttack() {
         return false;
     }
 
+    /**
+     * Mis-matching the token signature and token data, to verify if the JWT implementation verifies the signature properly.
+     * A malicious user can exploit this vulnerability by supplying an arbitrary claim in the JWT payload to obtain 
+     * new privileges or impersonate other users
+     *
+     * @throws JWTException
+     */
+    private boolean executeIncorrectSignatureAttack() {
+        try {
+            JWTHolder cloneJWTHolder = new JWTHolder(this.serverSideAttack.getJwtHolder());
+            JSONObject payloadJSONObject = new JSONObject(cloneJWTHolder.getPayload());
+            payloadJSONObject.put(INCORRECT_SIGNATURE_KEY, INCORRECT_SIGNATURE_VALUE);
+            cloneJWTHolder.setPayload(payloadJSONObject.toString());
+
+            if (this.serverSideAttack.getJwtActiveScanRule().isStop()) {
+                return false;
+            }
+
+            if (verifyJWTToken(cloneJWTHolder.getBase64EncodedToken(), serverSideAttack)) {
+                raiseAlert(
+                        MESSAGE_PREFIX,
+                        VulnerabilityType.INCORRECT_SIGNATURE,
+                        Alert.RISK_HIGH,
+                        Alert.CONFIDENCE_HIGH,
+                        cloneJWTHolder.getBase64EncodedToken(),
+                        serverSideAttack);
+                return true;
+            }
+            return false;
+        }
+        catch (JSONException ex) {
+            throw new JWTException(
```

Comment:
I think just logging the exception and returning false would be good in this case.